security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -478,6 +488,30 @@ func (sc *SecretController) scrtUpdated(oldObj, newObj interface{}) {
 
 	rootCertificate := sc.ca.GetCAKeyCertBundle().GetRootCertPem()
 
+	// Check if root certificate in key cert bundle is not up-to-date. With mutiple
+	// Citadel deployed in Istio, the root certificate in istio-ca-secret could be
+	// rotated by any Citadel and newer than the one in local key cert bundle.
+	if !bytes.Equal(rootCertificate, scrt.Data[RootCertID]) {
+		caSecret, scrtErr := sc.caSecretController.LoadCASecretWithRetry(CASecret,
+			sc.istioCaStorageNamespace, 100 * time.Millisecond, 5*time.Second)
+		if scrtErr != nil {
+			k8sControllerLog.Errorf("Fail to load CA secret %s:%s (error: %s), skip updating secret %s",
+				sc.istioCaStorageNamespace, CASecret, scrtErr.Error(), name)
+			return
+		}
+		// If root cert in workload secret matches istio-ca-secret, and does not match
+		// the root cert in key cert bundle. This implies that another Citadel has recently
+		// rotated root cert in istio-ca-secret and the workload secret. Reload key cert
+		// bundle and skip updating workload secret.
+		if bytes.Equal(scrt.Data[RootCertID], caSecret.Data[caCertID]) {
```

Comment:
I think we should always update the KeyCertBundle, the source of truth is the secret. So we shouldn't add this condition check here. WDYT?

Reply:
Also, please add a log here :)